### PR TITLE
Skip tests with missing requirements

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description
Skip tests with missing requirements

## Type of change
- [ ] Bug fix & code cleanup
- [ ] New feature
- [ ] Documentation update
- [x] Test update

## Checklist for the reviewer
This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

**Review by francescalb — CHANGES_REQUESTED** [review]
It is OK to skip tests locally due to missing of imports, but not on github. In this way we may risk to remove dependencies and thus stop testing parts of the code without it being picked up on in the CI/CD workflows. 

If this is to be accepted, we need to have an additional check after ctest, that check that all skipped tests are in the specified list of tests (e.g. mongodb-test cannot run on github)

**Comment by jesper-friis** [comment]
> It is OK to skip tests locally due to missing of imports, but not on github. In this way we may risk to remove dependencies and thus stop testing parts of the code without it being picked up on in the CI/CD workflows.
> 
> If this is to be accepted, we need to have an additional check after ctest, that check that all skipped tests are in the specified list of tests (e.g. mongodb-test cannot run on github)

It can be implemented with importskip() checking an environment variable. If it is set, the skip will not be accepted.

File: bindings/python/testutils.py [hunk 1]
```diff
@@ -71,15 +72,44 @@ def importcheck(module_name, package=None):
...
+    Notes:
+        If you want to run the tests and get an error if a module
+        cannot be imported, set environment variable
+        `DLITE_IMPORTSKIP_EXITCODE=1` before running the tests (if you
+        run with ctest, set `DLITE_IMPORTSKIP_EXITCODE` at configure
+        time).
+
+        For packages that depends on external services like postgresql,
+        call `importskip()` with `env_exitcode=None` to skip the test
+        regardless of `DLITE_IMPORTSKIP_EXITCODE` is set.
```
Comment (francescalb) [inline bindings/python/testutils.py:99]:
```suggestion
        For packages that depend on external services like postgresql,
        call `importskip()` with `env_exitcode=None` to skip the test
        regardless of whether `DLITE_IMPORTSKIP_EXITCODE` is set or not.
```

File: doc/user_guide/environment_variables.md [hunk 2]
```diff
@@ -96,6 +96,11 @@ DLITE_USE_BUILD_ROOT is set).
       - a glob pattern (/path/to/*.json)
     In the two last cases, the file extension must match the driver name.
 
+  - **DLITE_IMPORTSKIP_EXITCODE**: Exit code from tests that fails to
+    load a python module with `importskip()`.  Define this to 1 at
+    configure time if you want the tests to not be skipped if they
+    cannot import a needed module.
```
Comment (francescalb) [inline doc/user_guide/environment_variables.md:102]:
```suggestion
  - **DLITE_IMPORTSKIP_EXITCODE**: Exit code from tests that fail to
    load a python module with `importskip()`.  Define this to 1 at
    configure time if you want the tests not to be skipped if they
    cannot import a needed module.
```

**Review by francescalb — APPROVED** [review]
Very nice.

**Pull request merged**